kite-data/kite-data-core/src/main/java/org/kitesdk/data/spi/filesystem/CSVUtil.java

```diff
@@ -192,13 +216,26 @@ private static Schema inferSchemaInternal(String name, InputStream incoming,
           throw new DatasetException("Found null value for required field: " +
               fieldName + " (" + types[i] + ")");
         }
-        fieldAssembler = fieldAssembler.name(fieldName)
-            .doc("Type inferred from '" + sample(values[i]) + "'")
-            .type(schema(types[i], false)).noDefault();
+        if (specifiedTypes != null && specifiedTypes.containsKey(fieldName)) {
```

Comment:
If you take my suggestion above for a `doc` array instead of `values` then there is no need to do the field lookup twice and the logic is a bit simpler.